script/urlChecker.ts

```diff
@@ -0,0 +1,97 @@
+import chunk from 'lodash/chunk';
+import fetch from 'node-fetch';
+import ProgressBar from 'progress';
+import { argv } from 'yargs';
+import { content as contentRoute } from '../src/app/content/routes';
+import { Book, BookWithOSWebData, LinkedArchiveTreeSection } from '../src/app/content/types';
+import { findTreePages } from '../src/app/content/utils/archiveTreeUtils';
+import { getBookPageUrlAndParams, getUrlParamForPageId } from '../src/app/content/utils/urlUtils';
+import { assertDefined } from '../src/app/utils';
+import { findBooks } from './utils/bookUtils';
+
+const {
+  rootUrl,
+  bookId,
+  bookVersion,
+  archiveUrl,
+  useUnversionedUrls,
+} = argv as {
+  rootUrl?: string;
+  bookId?: string;
+  bookVersion?: string;
+  archiveUrl?: string;
+  useUnversionedUrls?: boolean;
+};
+
+async function checkPages(bookSlug: string, pages: string[]) {
+  let anyFailures = false;
+  const bar = new ProgressBar(`checking ${bookSlug} [:bar] :current/:total (:etas ETA)`, {
+    complete: '=',
+    incomplete: ' ',
+    total: pages.length,
+    width: 20,
+  });
+
+  const notFound: string[] = [];
+
+  const visitPage = async(page: string) => {
+    try {
+      const response = await fetch(`${rootUrl}${page}`);
+      if (response.status === 404) {
+        notFound.push(page);
+      }
+    } catch {
+      anyFailures = true;
+      bar.interrupt(`- (error loading) ${page}`);
+    }
+    bar.tick();
+  };
+
+  for (const pageChunk of chunk(pages, 50)) {
+    await Promise.all(pageChunk.map(visitPage));
+  }
+
+  if (notFound.length) {
+    console.log(`404'd: ${notFound.length === pages.length // tslint:disable-line:no-console
+      ? 'all'
+      : `\n${notFound.join('\n')}\n`}`);
+  }
+
+  return anyFailures;
```

Comment:
the other script was just looking for errors, but the purpose of this script is to check for 404ing urls, so we should `return anyFailures || notFound.length > 0` here